plugins/module_utils/ilo_redfish_utils.py

```diff
@@ -0,0 +1,247 @@
+            if uri != current_session:
+                response = self.get_request(self.root_uri + uri)
+                if response['ret'] is False:
+                    return response
+                data = response['data']
+                for property in properties:
+                    if property in data:
+                        session[property] = data[property]
+                sessions_results.append(session)
+        result["entries"] = sessions_results
+        result["ret"] = True
+        return result
+
+    def set_NTPServer(self, mgr_attributes):
+        result = {}
+        setkey = mgr_attributes['mgr_attr_name']
+
+        nic_info = self.get_manager_ethernet_uri()
+        ethuri = nic_info["nic_addr"]
+
+        response = self.get_request(self.root_uri + ethuri)
+        if(response['ret'] is False):
+            return response
+        result['ret'] = True
+        data = response['data']
+
+        # enable_value = "false"
+
+        payload = {"DHCPv4": {
+            "UseNTPServers": ""
+        }}
+
+        if(data["DHCPv4"]["UseNTPServers"] is True):
+            payload["DHCPv4"]["UseNTPServers"] = False
+            resDHV4 = self.patch_request(self.root_uri + ethuri, payload)
+            if resDHV4['ret'] is False:
+                return resDHV4
+
+        payload = {"DHCPv6": {
+            "UseNTPServers": ""
+        }}
+
+        if(data["DHCPv6"]["UseNTPServers"] is True):
+            payload["DHCPv6"]["UseNTPServers"] = False
+            resDHV6 = self.patch_request(self.root_uri + ethuri, payload)
+            if resDHV6['ret'] is False:
+                return resDHV6
+
+        DTuri = self.manager_uri + "DateTime"
+
+        response = self.get_request(self.root_uri + DTuri)
+        if(response['ret'] is False):
+            return response
+
+        data = response['data']
+
+        NTPlist = data[setkey]
+        if(len(NTPlist) == 2):
+            NTPlist.pop(0)
+
+        NTPlist.append(mgr_attributes['mgr_attr_value'])
+
+        payload = {}
+        payload[setkey] = NTPlist
+
+        response1 = self.patch_request(self.root_uri + DTuri, payload)
+        if response1['ret'] is False:
+            return response1
+
+        return {'ret': True, 'changed': True, 'msg': "Modified %s" % mgr_attributes['mgr_attr_name']}
+
+    def setTimeZone(self, attr):
+        # result = {}
+        key = attr['mgr_attr_name']
+
+        uri = self.manager_uri + "DateTime/"
+        response = self.get_request(self.root_uri + uri)
+        if(response['ret'] is False):
+            return response
+
+        data = response["data"]
+
+        if key not in data:
+            return {'ret': False, 'msg': "Key %s not found" % key}
+
+        TimeZonelist = data["TimeZoneList"]
+        index = ""
+
+        for i in range(0, len(TimeZonelist)):
+            if(attr['mgr_attr_value'] in TimeZonelist[i]["Name"]):
+                index = TimeZonelist[i]["Index"]
+                break
```

Comment:
More pythonic as:
```suggestion
        timezones = data["TimeZoneList"]
        index = ""

        for tz in timezones:
            if attr['mgr_attr_value'] in tz["Name"]:
                index = tz["Index"]
                break
```

Reply:
note that this suggestion, as-is, will break the code by renaming `TimeZonelist` to the more pythonic `timezones` name.